src/AdminCommands.ts

```diff
@@ -372,12 +372,15 @@ export class AdminCommands {
         return new Promise((resolve, reject) => {
             try {
                 let matched = false;
-                this.yargs.parse(argv, {
+                const possiblePromise = this.yargs.parse(argv, {
                     // eslint-disable-next-line @typescript-eslint/no-unused-expressions
                     completed: (err) => { err ? reject(err) : resolve(true); },
                     matched: () => { matched = true; },
                     respond,
                 }, (err) => {  if (err !== null) { reject(err); } });
+                if (possiblePromise instanceof Promise) {
+                    possiblePromise.catch((ex) => reject(ex));
```

Comment:
Give than we synchronously `resolve(false)` below this, I don't think we can rely on this actually catching the error?

Maybe a try-await would fit better here, just to add to a stew of having every single async mechanism in this method :D It is a bit nuts. Is this fixing some bug we've discovered?

Reply:
I think we've improved this immensely now

Reply:
We have indeed!